### PR TITLE
Add tooling support for SPV_KHR_maximal_reconvergence

### PR DESCRIPTION
**Pull request by alan-baker**
This depends on https://github.com/KhronosGroup/SPIRV-Headers/pull/407. I will update the deps after it is merged.

Adds tooling support for SPV_KHR_maximal_reconvergence
 * parsing and assembly
 * validation
 * optimizer pass to add/remove the execution mode from all entry points in a module

### CODE REVIEW

File: source/opt/modify_maximal_reconvergence.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+           IRContext::kAnalysisDecorations | IRContext::kAnalysisCombinators |
+           IRContext::kAnalysisCFG | IRContext::kAnalysisNameMap |
+           IRContext::kAnalysisConstants | IRContext::kAnalysisTypes;
+  }
+
+ private:
+  bool AddMaximalReconvergence();
+  bool RemoveMaximalReconvergence();
+
+  bool _add;
```
Comment (dneto0) [inline source/opt/modify_maximal_reconvergence.h:48]: nit: I would prefer a trailing underscore for a member variable., not a leading underscore.
That's the Google style guide recommendation ([here](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules:~:text=Data%20members%20of%20classes%2C%20both%20static%20and%20non%2Dstatic%2C%20are%20named%20like%20ordinary%20nonmember%20variables%2C%20but%20with%20a%20trailing%20underscore.))

For me, a leading underscore is close enough to me to confuse with reserved identifiers in C and C++

**Review by dneto0 — APPROVED** [review]
Needs the DEPS update to make the bots pass.
But LGTM!

File: source/opt/modify_maximal_reconvergence.h (hunk 1 above)
Comment (alan-baker) [reply source/opt/modify_maximal_reconvergence.h]: Made that change.

**Comment by sudonatalie** [comment]
It looks like you might have missed adding `--modify-maximal-reconvergence` to the help text in `opt.cpp`. (Just noticed when resolving merge conflicts rebasing onto this.)

**Pull request merged**